### PR TITLE
Maintain autosave timers while disconnected

### LINKED ISSUE #16798: fix for report of auto-save not working?
hello - did the fix proposed in https://discourse.jupyter.org/t/auto-save-does-not-always-work/24756 ever get submitted (and merged into) JupyterLab?

I have gotten a similar report from a user on one of our Jupyter instances (JupyterLab 4.2.4, JupyterHub 5.1.0), though we have not been able to reproduce the issue.

### PR DESCRIPTION
**Pull request by holzman**
## References
 
Fixes #16798.

## Code changes

The autosave feature works by starting a timer.  When the timer expires, it clears itself; then, if the session is connected, it attempts a save (which then restarts the timer for the next save).  However, if the session is disconnected, the timer never gets restarted - this code just calls `_setTimer()` in that case.

### CODE REVIEW

**Comment by holzman** [comment]
Added a test.

**Comment by JasonWeill** [comment]
@krassowski All tests are now passing. OK to merge and backport to 4.3.x?

**Comment by kellyrowland** [comment]
This should get backported further back than 4.3.x, no?

**Comment by JasonWeill** [comment]
We typically only backport changes further upon request. @kellyrowland How far back should we backport this fix once it's been merged into main?

**Comment by kellyrowland** [comment]
Ideally as far back as the version where the bug was introduced, but I would settle for 4.2.x 🙂

File: packages/docmanager/test/savehandler.spec.ts
Comment (krassowski) [inline packages/docmanager/test/savehandler.spec.ts:65]: Ideally tests should interface with the public API rather than private implementation details, as the implementation may change more rapidly and the test will need modifying (or worse would get deleted). That said I do not want to hold this off so if anyone else is happy to approve as-is I will not block it.

Comment (holzman) [reply packages/docmanager/test/savehandler.spec.ts]: Good point - I've revised the test.

**Comment by holzman** [comment]
BTW, while poking around the docmanager tests, I found one that wasn't really testing anything and fixed it in #16933.

**Review by krassowski — APPROVED** [review]
Thank you @holzman!

**Comment by kellyrowland** [comment]
Is there an ETA for when these backports will get a release? We'd really like to get the fix out to our users.

**Comment by JasonWeill** [comment]
@jupyterlab/release Per comment above, is now an appropriate time to cut new 4.2.x and 4.3.x releases to include backports of this fix?

**Comment by krassowski** [comment]
Yes 4.3.x is on the menu today. Not sure if 4.2.x is ready to go I will let @JasonWeill make the call.

**Comment by krassowski** [comment]
As of today it looks like this would be the only user-facing change on 4.2.6: https://github.com/jupyterlab/jupyterlab/compare/v4.2.5...4.2.x.

**Comment by JasonWeill** [comment]
@kellyrowland Version 4.2.6 is now available, which includes this change: https://github.com/jupyterlab/jupyterlab/releases/tag/v4.2.6

**Comment by kellyrowland** [comment]
Thank you!

**Comment by JasonWeill** [comment]
[v4.3.1](https://github.com/jupyterlab/jupyterlab/releases/tag/v4.3.1) is now also released, and that also includes this fix. Thank you @krassowski !

**Pull request merged**
